### PR TITLE
Migrate run generator check to Prefect

### PR DESCRIPTION
**Pull request by LucasG0**
Targeting `develop` as it involves removing current check / validator logic relying on redis
I also manually tested a proposed changed using demo-edge repository

### CODE REVIEW

File: backend/infrahub/proposed_change/tasks.py [hunk 1]
```diff
@@ -644,3 +648,144 @@ def _should_render_artifact(artifact_id: str | None, managed_branch: bool, impac
...
+
+@flow(
+    name="run-generator-as-check",
+    flow_run_name="Execute Generator {model.generator_definition.definition_name} for {model.target_name}",
+)
+async def run_generator_as_check(model: RunGeneratorAsCheckModel, service: InfrahubServices) -> ValidatorConclusion:
+    if model.proposed_change:
+        await add_tags(branches=[model.branch_name], nodes=[model.proposed_change], db_change=True)
+    else:
+        await add_tags(branches=[model.branch_name], nodes=[model.repository_id], db_change=True)
```
Comment (ogenstad) [inline backend/infrahub/proposed_change/tasks.py]: I wonder if we should always run this second part to tie it to the repository.

File: backend/infrahub/proposed_change/tasks.py [hunk 2]
```diff
@@ -644,3 +648,144 @@ def _should_render_artifact(artifact_id: str | None, managed_branch: bool, impac
...
+                "origin": model.repository_id,
+                "kind": "GeneratorDefinition",
+                "validator": model.validator_id,
+                "created_at": Timestamp().to_string(),
+                "message": check_message,
+                "conclusion": conclusion.value,
+                "instance": generator_instance.id,
+            },
+        )
+        await check.save()
```
Comment (ogenstad) [inline backend/infrahub/proposed_change/tasks.py:749]: Not related to this PR but this pattern is repeated in a number of places, i.e

1. Search of a check
2. Update if it exists
3. Create a new check if it doesn't

We have something similar in start_validator() we should probably clean this up and remove some of the boilerplate.

**Review by ogenstad — COMMENTED** [review]
LGTM, I'll also run a quick test.

File: backend/infrahub/proposed_change/tasks.py (hunk 1 above)
Comment (LucasG0) [reply backend/infrahub/proposed_change/tasks.py]: Right, actually this path never runs as `model.proposed_change` is never `None` here

**Review by ogenstad — APPROVED** [review]
